### DISCUSSION TITLE
How to debug indexing related issues?

### DISCUSSION THREAD

**Discussion by mattkallo** [milvus-io/milvus#24067] [Q&A and General discussion] [answered]
I am trying to index around 2000 documents, but it doesn't seem to be creating the index. I saw there is a min for index creation, which is 1024 records.  Could you pls check the below and let me know what could be wrong?

```
collection.num_entities
1806

index_params = {
  "metric_type":"L2",
  "index_type":"IVF_FLAT",
  "params":{"nlist":100}
}

collection.create_index(
  field_name="contentVector", 
  index_params=index_params
)
Status(code=0, message=)

utility.index_building_progress("search_index")
{'total_rows': 0, 'indexed_rows': 0}
```

**Comment by yhmo** [comment]
The index has been created after you called. create_index(). Since the row count (1806) is few, the index was created quickly. The utility.index_building_progress() shows "total_rows: 0", this is a tech limitation of this interface, it cannot detect the segments generated before create_index().

Actually, the index has been successfully built.

**Reply by mattkallo** [reply]
Thanks for the clarification. Another related question, do we have to re-create/rebuild the index when more data is loaded?

**Reply by yhmo** [reply, marked as answer]
Milvus manages data by segments. When you continually insert data, the new data is accumulated in a buffer. After the data size exceeds a threshold(by default 512MB*0.23), milvus will flush the buffer into a new segment.
Once you have called create_index(), milvus will automatically build index for new segments. No need to call create_index() again.
